src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -3813,5 +3969,15 @@ Node TheoryStringsRewriter::returnRewrite(Node node, Node ret, const char* c)
 {
   Trace("strings-rewrite") << "Rewrite " << node << " to " << ret << " by " << c
                            << "." << std::endl;
+  // standard post-processing
+  if (ret.getKind() == EQUAL && node.getKind() != EQUAL)
```

Comment:
Do the same for conjunction of equalities (e.g. `(and (= x y) (= z w))`).

Reply:
The slightly annoying thing here is that technically we should do it for any Boolean combination of equalities.